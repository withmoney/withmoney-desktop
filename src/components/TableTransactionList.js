import React from 'react';
import PropTypes from 'prop-types';
import TransactionsItem from './TableTransactionItem';

const TransactionsList = ({ list }) => {
  if (!list.length) {
    return null;
  }

  return list.map(item => <TransactionsItem key={item.id} transaction={item} />);
};

TransactionsList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      transactionDate: PropTypes.string,
      name: PropTypes.string,
      CategoryId: PropTypes.number,
      value: PropTypes.string,
    }),
  ),
};

TransactionsList.defaultProps = {
  list: [],
};

export default TransactionsList;
