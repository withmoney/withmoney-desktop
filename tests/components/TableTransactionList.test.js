import renderer from 'react-test-renderer';
import TableTransactionList from 'components/TableTransactionList';

jest.mock('components/TableTransactionItem', () => 'table-transaction-item');

describe('TableTransactionList', () => {
  let Component;

  beforeAll(() => {
    Component = global.setup(TableTransactionList);
  });

  it('should render null when dont have any transaction', () => {
    const wrapper = renderer.create(Component());

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render spin when loading status is true', () => {
    const wrapper = renderer.create(
      Component({
        isLoading: true,
      }),
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render table with list of transactions', () => {
    const wrapper = renderer.create(
      Component({
        list: [
          {
            id: 99,
            name: 'Name',
            transactionDate: '2018-08-01 20:80',
            CategoryId: 1,
            value: '10.00',
          },
        ],
      }),
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
