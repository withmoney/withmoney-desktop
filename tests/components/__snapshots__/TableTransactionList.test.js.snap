// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableTransactionList should render null when dont have any transaction 1`] = `
<div
  className="table-transactions__no-data"
>
  No Transaction on this month.
</div>
`;

exports[`TableTransactionList should render spin when loading status is true 1`] = `
<div
  className="table-transactions__loading"
>
  <img
    alt="spin"
    className="spin"
    src="/static/img/spin-black.svg"
  />
</div>
`;

exports[`TableTransactionList should render table with list of transactions 1`] = `
<table-transaction-item
  columns={
    Array [
      Object {
        "label": "Is Paid?",
        "name": "isPaid",
        "style": Object {
          "flexBasis": "15%",
        },
      },
      Object {
        "label": "Date",
        "name": "transactionDate",
        "style": Object {
          "flexBasis": "15%",
        },
      },
      Object {
        "label": "Name",
        "name": "name",
        "style": Object {
          "flexBasis": "20%",
        },
      },
      Object {
        "label": "Category",
        "name": "CategoryId",
        "style": Object {
          "flexBasis": "20%",
        },
      },
      Object {
        "label": "Value",
        "name": "value",
        "style": Object {
          "flexBasis": "15%",
        },
      },
      Object {
        "label": "Action",
        "name": "action",
        "style": Object {
          "flexBasis": "15%",
        },
      },
    ]
  }
  transaction={
    Object {
      "CategoryId": 1,
      "id": 99,
      "name": "Name",
      "transactionDate": "2018-08-01 20:80",
      "value": "10.00",
    }
  }
/>
`;
